src/util/cilfacade.ml

```diff
@@ -77,10 +78,29 @@ class allBBVisitor = object (* puts every instruction into its own basic block *
   method! vtype _ = SkipChildren
 end
 
+class extractConstantsVisitor(list) = object
+  inherit nopCilVisitor
+
+  method! vexpr e =
+    match e with
+    | Const (CInt64(i,ik,_)) ->
+      (if (Int64.compare i (Int64.of_int min_int) >= 0) && (Int64.compare i (Int64.of_int max_int)) <= 0 then
```

Comment:
What do these comparisons really guarantee? If inside `CInt64` the integer component already overflowed, it might have wrapped around into this supposedly valid range still and you couldn't detect it after the fact.

It wouldn't be any more difficult to just handle the string component case as well (and use bigints) and ensure no such problem can occur.

Reply:
That's true, they don't guarantee much here. The reason I did not go with bigints here was that Apron in the end will only accept `int` anyway, but I guess you're right and one should use BigInts here.

Reply:
> The reason I did not go with bigints here was that Apron in the end will only accept `int` anyway

It accepts bigints through its own GMP interface just fine, just not the Zarith ones:
https://github.com/goblint/analyzer/blob/791b561ea789f06b084cbb752b3349e38f5cd3a3/src/cdomains/apron/apronDomain.apron.ml#L105-L110

If this mechanism is to be general, the general list (or really a set), should be of our Zarith bigints anyway and apron will have to do the conversion to use them.

Reply:
I moved it into a separate module now and made it more generic